functions/CheckForCustomer.js

```diff
@@ -53,72 +59,110 @@ let CheckForCustomer = function (ncUtil,
+        if (!err) {
+          client.ReadMultiple(args, function(error, body, envelope, soapHeader) {
+            if (!error) {
+              if (!body.ReadMultiple_Result) {
+                // If ReadMultiple_Result is undefined, no results were returned
+                out.ncStatusCode = 204;
+              } else if (Array.isArray(body.ReadMultiple_Result[customerServiceName])) {
+                // If an array is returned, multiple customers were found
+                out.ncStatusCode = 409;
+                out.payload.error = { err: body };
+              } else if (typeof body.ReadMultiple_Result[customerServiceName] === 'object') {
+                // If an object is returned, one customer was found
+                out.ncStatusCode = 200;
+                out.payload = {
+                  customerRemoteID: body.ReadMultiple_Result[customerServiceName].No,
+                  customerBusinessReference: nc.extractBusinessReference(channelProfile.customerBusinessReferences, body.ReadMultiple_Result)
+                };
+              } else {
+                // Unexpected case
+                out.ncStatusCode = 400;
+                out.payload.error = { err: body };
+              }
+              callback(out);
             } else {
-              out.ncStatusCode = 204;
+              if (error.response) {
+                logError("Error - Returning Response as 400 - " + error, ncUtil);
+                out.ncStatusCode = 400;
+                out.payload.error = { err: error };
+                callback(out);
+              } else {
+                logError("CheckForCustomer Callback error - " + error, ncUtil);
+                out.ncStatusCode = 500;
+                out.payload.error = { err: error };
+                callback(out);
+              }
             }
-          } else if (response.statusCode == 429) {
-            out.ncStatusCode = 429;
-            out.payload.error = body;
-          } else if (response.statusCode == 500) {
-            out.ncStatusCode = 500;
-            out.payload.error = body;
-          } else {
+          });
+        } else {
+          let errStr = String(err);
+
+          if (errStr.indexOf("Code: 401") !== -1) {
+            logError("401 Unauthorized (Invalid Credentials) " + errStr);
             out.ncStatusCode = 400;
-            out.payload.error = body;
+            out.response.endpointStatusCode = 401;
+            out.response.endpointStatusMessage = "Unauthorized";
+          } else {
+            logError("CheckForCustomer Callback error - " + err, ncUtil);
+            out.ncStatusCode = 500;
+            out.payload.error = { err: err };
           }
           callback(out);
-        } else {
-          // If an error occurs, log the error here
-          logError("Do CheckForCustomer Callback error - " + error, ncUtil);
-          out.ncStatusCode = 500;
-          out.payload.error = error;
-          callback(out);
         }
       });
     } catch (err) {
```

Comment:
This try/catch may not work as expected, it will only catch a synchronous error, but the code within it is all asynchronous.

Looking now though it is like this in the template as well and should probably be fixed there too.